version: '3.8'

services:
  frontend:
    build:
      context: ./etec-pi-2024-frontend-Santiii4-main
      dockerfile: Dockerfile
    ports:
      - "3000:5000"
    networks:
      - app-network

  backend:
    build:
      context: ./etec-pi-2024-backend-Santiii4-main
      dockerfile: Dockerfile
    ports:
      - "5000:5000"
    environment:
      MYSQLDB_HOST: mysqldb
      MYSQLDB_ROOT_PASSWORD: 123456
      MYSQLDB_DOCKER_PORT: 3306
      MYSQLDB_DATABASE: ecommerce
    depends_on:
      mysqldb:
        condition: service_healthy
    networks:
      - app-network

  mysqldb:
    image: mysql:8.0
    environment:
      MYSQL_ROOT_PASSWORD: 123456
      MYSQL_DATABASE: ecommerce
    ports:
      - "3307:3306"
    volumes:
      - db_data:/var/lib/mysql
      # La siguiente l√≠nea ha sido comentada para evitar el conflicto de montaje
      # - ./etec-pi-2024-backend-Santiii4-main/init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD-SHELL", "mysqladmin ping -h localhost -u root -p123456 --silent"]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  db_data:






